### PR TITLE
Adding PubGuru to amp-ad

### PR DESCRIPTION
**Pull request by PubGuru**
Adding [Pubguru](https://www.pubguru.com) AMP support

### CODE REVIEW

**Comment by yayolius** [comment]
I signed it!

File: package.json [hunk 1]
```diff
@@ -79,7 +79,7 @@
     "fetch-mock": "5.13.1",
     "formidable": "1.1.1",
     "fs-extra": "5.0.0",
-    "gulp": "3.9.1",
+    "gulp": "^3.9.1",
```
[the remark this answers is not in the document]
Comment (PubGuru) [reply package.json]: Yes no problem, i don't know how it was changed

**Comment by PubGuru** [comment]
I updated the line. Thanks.

**Comment by lannka** [comment]
@PubGuru could you follow the CLA instruction?

**Comment by lannka** [comment]
Please also consider impemement render-start API. see https://github.com/ampproject/amphtml/blob/master/ads/README.md#other-tips

**Comment by PubGuru** [comment]
hi @lannka, this was done by my boss. about the renderStart we just used the way that other adapter where creating their adapters, but if this is necessary there is no problem.

**Comment by lannka** [comment]
we highly recommend the usage of renderstart, which gives a much better ad loading experience.

was your CLA signed as a corp or as individual? if it's a corp, please list the corp name here.

**Comment by PubGuru** [comment]
@lannka yes, the CLA is signed to a corporation is MonetizeMore.com it is signed with the google.contibutors@monetizemore.com email.

**Comment by lannka** [comment]
@PubGuru I couldn't find google.contibutors@monetizemore.com in our CLA list. can you double check?

**Comment by PubGuru** [comment]
the CLA is signed as "Esell Solutions Ltd. DBA MonetizeMore"
Also please check this email claudio@monetizemore.com.
Thanks

**Comment by PubGuru** [comment]
Ok, seems that  CLA is passed properly, it is mandatory to implement the renderStart?, i can add it to our adapter code but it won't show either way in this pull request.

Thanks

**Comment by lannka** [comment]
@PubGuru if you've already implemented renderStart, please put `renderStartImplemented: true,` in `_config.js`

**Comment by PubGuru** [comment]
Ok, we implemented renderStart and checked it on our side. Hope we can get approval soon.
Ty

File: ads/pubguru.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
```
Comment (zhouyx) [inline ads/pubguru.js]: nit: 2018

File: ads/pubguru.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function pubguru(global, data) {
+  validateData(data, ['publisher', 'slot']);
+
+  global.$pubguru = data;
+
+  const el = global.document.createElement('div');
+  el.setAttribute('id','the-ad-unit');
```
Comment (zhouyx) [inline ads/pubguru.js]: nit: space after comma

File: ads/pubguru.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
+<!---
+Copyright 2016 The AMP HTML Authors. All Rights Reserved.
```
Comment (zhouyx) [inline ads/pubguru.md]: nit: 2018

File: ads/_config.js [hunk 5]
```diff
@@ -642,6 +642,10 @@ export const adConfig = {
 
   pubexchange: {},
 
+  pubguru: {
+    renderStartImplemented: true,
```
Comment (zhouyx) [inline ads/_config.js]: Could you confirm that the `renderStart` API is rolled out? When I try to test with your example, I couldn't receive the `render-start` message. Note that AMP runtime won't display an ad before that message is received or timeout.

Comment (PubGuru) [reply ads/_config.js]: Sorry the issue is that i was only calling renderStart when we had a winning ad, but forgot to implement the test version, it is updated. Please check again. Thanks

**Comment by PubGuru** [comment]
hi @zhouyx any update with this? thanks

**Comment by PubGuru** [comment]
Sorry to insist but is there anything that we can do to speed up the process?
Thanks

**Comment by zhouyx** [comment]
Hey @PubGuru sorry about the delay. But I still couldn't get the `render-start` message when testing with your ad example. Could you double check, and confirm that it works locally in our test environment. Thanks.
Other than that the code looks good. However you may want to rebase to latest master to fix the percy test.

**Comment by PubGuru** [comment]
Hey @zhouyx thanks, how can i check for that render-start message?
Doing the rebase now. Thanks

**Comment by zhouyx** [comment]
Simply check the post messages from the ad iframe to the parent doc with whatever way you like. I always go with logging out received message here https://github.com/ampproject/amphtml/blob/master/extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js#L204

**Comment by PubGuru** [comment]
ok, just did some changes and printed the postmessages, i am getting:
captura de pantalla 2018-04-02 a las 4 46 20 p m

**Comment by zhouyx** [comment]
@PubGuru Get `render-start` now. Thanks. I've restarted the travis test, hopefully it will fix the flaky test. Thanks.

**Comment by PubGuru** [comment]
@zhouyx can you help me? i did a pull rebase from the source, because the test was failing, i am really lost right now, it seems that the rebase removed your approval and the CLA check.

what do you think i need to do to solve this?

Thanks
Claudio

**Comment by zhouyx** [comment]
@PubGuru There's some issue with the rebase. I imagine it is because you pull upstream master from the working branch, and rebase to local master. 

Please make sure you pull from upstream master to your local master, and then rebase from your working branch to your local master

Please refer to the instructions [here](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-quick.md#pull-the-latest-changes). Thanks

**Comment by PubGuru** [comment]
@lannka  or @zhouyx can you reaprove the cla please?
thanks

**Comment by zhouyx** [comment]
Hi @PubGuru @josh313 We don't approve CLA, it is updated automatically.

CLA is missing because the PR is not rebased correctly. 
Please refer to my [comment](https://github.com/ampproject/amphtml/pull/13461#issuecomment-380590442) above

If rebased correctly, you won't be able to see other's commits in the PR.
image

**Comment by PubGuru** [comment]
Hi,ok we finally solved the merging issues, our changes are very minimal, but it seems that the test keeps failing for external issues not related to the code that we deployed.
What else can we do at this point?

**Comment by cathyxz** [comment]
I restarted the tests (you can technically restart CI by yourself too with a no-op `commit --amend`). Leaving it to @zhouyx to merge your changes whenever CI turns green. =)

**Comment by PubGuru** [comment]
@cathyxz @cathyxz @lannka sorry to bother you, but i have several clients that wan't to use this adapter i tried to do build my own version from master (but it reloads the amp-ad file to: https://cdn.ampproject.org/rtv/011524862012167/v0/amp-ad-0.1.js).

I was also waiting for the cdn to get updated but i don't know the time tables to that.
in the meantime i am doing tests in here: http://35.153.232.63/ but nothing has worked until now.
Thanks.

**Comment by cathyxz** [comment]
@PubGuru we release on a 2 week schedule. Your changes are in https://github.com/ampproject/amphtml/releases/tag/1525302057221. I think it will spend a week in canary before going to production. You can use/test with canary by opting in to our dev channel (for your browser only) by turning it on here: https://cdn.ampproject.org/experiments.html

You can find more information on releases at: 
https://github.com/ampproject/amphtml/blob/master/contributing/release-schedule.md

**Comment by PubGuru** [comment]
Thanks @cathyxz

**Pull request merged**